app/models/user.rb

```diff
@@ -173,15 +175,17 @@ def remove_user_from_kudos
     Kudo.where(user: self).update_all(user_id: nil)
   end
 
-  def log_removal_as_next_of_kin
+  def log_removal_of_self_as_next_of_kin
     fannish_next_of_kins.each do |fnok|
-      fnok.user.create_log_item({
-                                  action: ArchiveConfig.ACTION_REMOVE_FNOK,
-                                  fnok_user_id: self.id
-                                })
+      log_removal_of_next_of_kin(fnok.user, self)
     end
   end
 
+  def log_removal_of_self_as_previous_of_kin
```

Comment:
I get why it's `previous_of_kin`, but I am not sure it is intuitively used in English. I would go with putting the two removal steps in one callback, because the named variables have helped to improve the clarity.